sdk/metricsadvisor/azure-ai-metricsadvisor/src/main/java/com/azure/ai/metricsadvisor/MetricsAdvisorClientBuilder.java

```diff
@@ -178,9 +181,12 @@ private HttpPipeline getDefaultHttpPipeline(Configuration buildConfiguration) {
         final List<HttpPipelinePolicy> policies = new ArrayList<>();
 
         // Authentications
-        if (credential != null) {
-            headers.put(OCP_APIM_SUBSCRIPTION_KEY, credential.getSubscriptionKey());
-            headers.put(API_KEY, credential.getApiKey());
+        if (tokenCredential != null) {
+            policies.add(new BearerTokenAuthenticationPolicy(tokenCredential, DEFAULT_SCOPE));
+        } else if (!CoreUtils.isNullOrEmpty(metricsAdvisorKeyCredential.getSubscriptionKey())
+            || !CoreUtils.isNullOrEmpty(metricsAdvisorKeyCredential.getApiKey())) {
```

Comment:
not sure we need to be consistent across languages on this condition to set headers, but just want to point it. LGTM otherwise.

Reply:
Confirmed with other langs and this is the expected behavior.